### PR TITLE
Fixes RMCE bugs

### LINKED ISSUE #3955: [Bug] Rapid machinery component exchanger issue
Filling up parts to the RMCE with a gadget bag doesn't remove the parts from the gadget bag so you get two copies of each part, one in the RMCE and one in the bag.

The delays for the RMCE working need to be shortened if possible.

### LINKED ISSUE #3960: (WEB REPORT BY: kurfurst REMOTE: 198.245.63.50:7777) RCME Issues
>Revision 
a1d008abaa220a4d351d0db18259d99708dd7766 

>General description of the issue
The RCME immediately kicks out Super Capacitors and Super Matter Bins when they are inserted. However, it still counts them as stocked, meaning that you could for instance repeatedly insert one Super Matter Bin and then replace 3 matter bins with only 1.

As a matter of preference, the RMCE should also display what's inside when examined, or be able to be opened to check inside like a pill collector or gadget bag.

### PR DESCRIPTION
**Pull request by Kurfursten**
Probably controversial because it pretty much rips out a bunch of snowflake menus that are no longer necessary.
- RMCE is now a child of storage, can quick grab and quick drop - so the eject menu is no longer necessary
- You can also just look inside like a bag, so no need for a 'scan internal components'
- You no longer need to wait for it to scan the machine a second time to 'output scanned data' - what was that first three seconds about if not scanning?
- There is no longer an item duping bug
- Fixes #3960
- Fixes #3955
- Also DNA Modifier compatible - removes snowflake copy/paste code

### CODE REVIEW

**Comment by 9600bauds** [comment]
:+1: you are doing god's work

**Comment by Probe1** [comment]
:+1: 

I will be nicer and compliment strangers if merged.

File: code/modules/research/mechanic/component_exchanger.dm [hunk 1]
```diff
@@ -16,41 +16,31 @@
...
-	var/maxcomponents = 20
 	var/emagged = 0 //So we can emag it for "improved" functionality
 
-//Do a quick check-up of what is loaded
-/obj/item/device/component_exchanger/attack_self(var/mob/user)
+	allow_quick_gather = 1
+	use_to_pickup = 1
+	allow_quick_empty = 1
+	storage_slots = 21
+	can_hold = new/list(/obj/item/weapon/stock_parts/)
```
Comment (ComicIronic) [inline code/modules/research/mechanic/component_exchanger.dm]: You don't need the new/, and all the elements of can_hold are strings, not plain paths.

File: code/modules/research/mechanic/component_exchanger.dm [hunk 2]
```diff
@@ -16,41 +16,31 @@
...
 			user << "<span class='warning'>\The [G] is empty.</span>"
 		for(var/obj/item/weapon/stock_parts/S in G.contents)
-			if(componentstorage.len < maxcomponents)
-				componentstorage += S
+			if(src.contents.len < storage_slots)
+				src.contents += S
 			else
 				user << "<span class='notice'>You fill \the [src] to its capacity with \the [G]'s contents.</span>"
 				return
 		user << "<span class='notice'>You fill up \the [src] with \the [G]'s contents.</span>"
```
Comment (ComicIronic) [inline code/modules/research/mechanic/component_exchanger.dm]: Be sure to return 1 after this

Comment (Kurfursten) [inline code/modules/research/mechanic/component_exchanger.dm]: done

**Pull request merged**